Line/android/LINELoginDemo/app/src/main/java/com/google/firebase/linelogindemo/activity/MainActivity.java

```diff
@@ -0,0 +1,151 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.linelogindemo.activity;
+
+import android.app.ProgressDialog;
+import android.content.DialogInterface;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.android.volley.toolbox.ImageLoader;
+import com.android.volley.toolbox.NetworkImageView;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.linelogindemo.R;
+import com.google.firebase.linelogindemo.util.LineLoginHelper;
+import com.google.firebase.linelogindemo.util.NetworkSingleton;
+
+public class MainActivity extends AppCompatActivity {
+
+    // Views
+    private Button mLineLoginButton;
+    private View mLoggedInView;
+    private TextView mDisplayNameText;
+    private NetworkImageView mProfileImageView;
+
+    // Helpers
+    private ImageLoader mImageLoader;
+    private LineLoginHelper mLineLoginHelper;
+
+    private final String TAG = this.getClass().getSimpleName();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        bindUIElements();
+
+        mImageLoader = NetworkSingleton.getInstance(this).getImageLoader();
+        mLineLoginHelper = new LineLoginHelper(this);
+        updateUI();
+    }
+
+    private void bindUIElements() {
+        // Bind views
+        mLineLoginButton = (Button) findViewById(R.id.line_login_button);
+        mLoggedInView = (View) findViewById(R.id.logged_in_view);
+        mDisplayNameText = (TextView) findViewById(R.id.display_name_text);
+        mProfileImageView = (NetworkImageView) findViewById(R.id.profile_image_view);
+
+        // Bind buttons' OnClickListener
+        mLineLoginButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                onTapLineLogin();
+            }
+        });
+        Button logoutButton = (Button) findViewById(R.id.logout_button);
+        logoutButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                onTapLogout();
+            }
+        });
+    }
+
+    void onTapLineLogin() {
+        // Show loading dialog
+        final ProgressDialog dialog = new ProgressDialog(this);
```

Comment:
You'll want to store this dialog as an instance variable and make sure to dismiss it (if it's showing) in `onStop()`, otherwise you'll get some nasty logs if the user quits while this is executing.